features/nvstore/mbed_lib.json

```diff
@@ -0,0 +1,31 @@
+{
+    "name": "nvstore",
+    "config": {
+        "enabled": {
+            "macro_name": "NVSTORE_ENABLED",
+            "value": true,
+            "help": "Enabled"
+        },
+        "max_keys": {
+            "macro_name": "NVSTORE_MAX_KEYS",
+            "value": 16,
+            "help": "Maximal number of allowed NVStore keys"
+        },
+        "area_1_address": {
+            "macro_name": "NVSTORE_AREA_1_ADDRESS",
```

Comment:
When area addresses are passed in by the user, how are these validated for sector alignment lack of overlap with area 2?

Reply:
Please observe the the calc_validate_area_params function: When calculating the sector map, it has the following piece of code:
```
        if (user_config) {
            // User configuration - here we validate it
            // Check that address is on a sector boundary, that size covers complete sector sizes,
            // and that areas don't overlap.
            if (_flash_area_params[area].address == flash_addr) {
                in_area = 1;
            }
            if (in_area) {
                area_size += sector_size;
                if (area_size == _flash_area_params[area].size) {
                    area++;
                    if (area == NVSTORE_NUM_AREAS) {
                        break;
                    }
                    in_area = 0;
                    area_size = 0;
                }
            }
        }
```
This code catches the case where we reach an NVStore area (starting in a sector start), and completes the check when the configured area size equals the sum of the consecutive sectors. If this condition is fulfilled, we increment the number of areas by 1, moving to the next area. Then after this loop ends, we check that the number of areas is exactly 2:
```
    if (user_config) {
        // Valid areas were counted. Assert if not the expected number.
        MBED_ASSERT(area == NVSTORE_NUM_AREAS);
```
Now, if the two areas overlap, the number of areas will never reach 2, as we first need to exit the first area for it to happen.

Reply:
Thx.
1. What is the user facing error message when the the area address is not on a sector boundary?
2. What is the user facing error message when the the area size is not equal to the size of the area address?

Reply:
Both will be asserted by the ```MBED_ASSERT(area == NVSTORE_NUM_AREAS);``` line.
I can print a clearer message (followed by an ```MBED_ASSERT(0)```) telling the user to make sure that the addresses are aligned to sector boundaries et al.